evalml/pipelines/utils.py

```diff
@@ -1508,22 +1503,28 @@ def stack_X(X, series_id_name, time_index, starting_index=None, series_id_values
         time_index (str): The name of the time index column.
         starting_index (int): The starting index to use for the stacked DataFrame. If None, the starting index
             will match that of the input data. Defaults to None.
-        series_id_values (set, list): The unique values of a series ID, used to generate the index. If None, values will
+        series_id_values (list): The unique values of a series ID, used to generate the index. If None, values will
             be generated from X column values. Required if X only has time index values and no exogenous values.
             Defaults to None.
 
     Returns:
         pd.DataFrame: The restacked features.
     """
     original_columns = set()
-    series_ids = series_id_values or set()
-    if series_id_values is None:
+    if series_id_values is not None:
+        series_ids = series_id_values
+    else:
+        # Using list to maintain order (vs. a set)
+        series_ids = list()
         for col in X.columns:
             if col == time_index:
                 continue
             separated_name = col.split(MULTISERIES_SEPARATOR_SYMBOL)
             original_columns.add(MULTISERIES_SEPARATOR_SYMBOL.join(separated_name[:-1]))
-            series_ids.add(separated_name[-1])
+            series_ids.append(separated_name[-1])
+        # Remove duplicates
+        seen = set()
+        series_ids = [val for val in series_ids if not (val in seen or seen.add(val))]
```

Comment:
We can simplify this to `series_ids = list(set(series_ids))` - converting to a set automatically removes duplicates

Reply:
Ah I see, makes sense. Do we care about insertion order? (I assume we do, but I'm not sure why) It makes sense to do it this way then, but a comment calling this out is certainly helpful

Reply:
Yep, when we restack the X columns we need it to be the correct order. It's used in line 1547 to generate the stacked X series ID values.